packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1522,6 +1542,20 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    private processAliasMessage(
+        message: ISequencedDocumentMessage,
+        localOpMetadata: unknown,
+        local: boolean,
+    ) {
+        type PendingAliasResolve = (value: IDataStoreAliasMapping | undefined) => void;
```

Comment:
It's a bit weird to see type definition here (given that some other place should be using same type to push that local metadata in the first place). Can we push it up and use in that other place to make sure compiler helps us ensure code stays in sync for producer and consumer of that data?

Reply:
Maybe other way is better - why not pass localOpMetadata to processAliasMessage() and let that layer own everything?
I think it will be a bit cleaner, and you will not need IDataStoreAliasMapping interface.

Reply:
I do not see any code that submits ContainerMessageType.AssignAlias ops in this PR (other than test code).
So, it's hard to evaluate and check if the flow (of storing callback in such form) make sense. Any particular reason you decided not to include it here? At least as a building block (maybe not used otherwise yet)?

Reply:
> I do not see any code that submits ContainerMessageType.AssignAlias ops in this PR 

That is intentional as to not expose the functionality. Currently the goal here is to prepare older versions of the runtime handle the op properly, hence it's only submitted in the test code.

I'll look into the localOpMetadata refactoring..